subcommands/waves/init.go

```diff
@@ -0,0 +1,132 @@
+package waves
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/docker/go/canonical/json"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/foundriesio/fioctl/client"
+	"github.com/foundriesio/fioctl/subcommands"
+	tuf "github.com/theupdateframework/notary/tuf/data"
+)
+
+func init() {
+	initCmd := &cobra.Command{
+		Use:   "init <wave> <version> [<tag>]",
+		Short: "Create a new wave for a given targets version.",
+		Long: `Create a new wave for a given targets version.",
+Optionally, provide a new tag to use for these targets ('master' by default).
+In any case, original tags of these targets set before creating a wave are ignored.
+
+This command only initializes a wave, but does not provision its updates to devices.
+Use a "fioctl wave rollout <wave> <group>" to trigger updates of this wave to one group.
+Use a "fioctl wave promote <wave>" to update all devices (make it globally available).`,
+		Run:  doInitWave,
+		Args: cobra.RangeArgs(2, 3),
+	}
+	cmd.AddCommand(initCmd)
+	initCmd.Flags().IntP("expires-days", "e", 0, `Role expiration in days; default 90.
+The same expiration will be used when a wave is promoted.`)
+	initCmd.Flags().StringP("expires-at", "E", "", `Role expiration date and time in RFC 3339 format.
+The same expiration will be used when a wave is promoted.
+When set this value overrides an 'expires-at' argument.
+Example: 2020-01-01T00:00:00Z`)
+	initCmd.Flags().BoolP("dry-run", "d", false, "Don't create a wave, print it to standard output.")
+	// TODO: Add an optional offline signing arguments
+}
+
+func doInitWave(cmd *cobra.Command, args []string) {
+	factory := viper.GetString("factory")
+	name, version, tag := args[0], args[1], "master"
+	if len(args) > 2 {
+		tag = args[2]
+	}
+	expires := readExpiration(cmd)
+	dryRun, _ := cmd.Flags().GetBool("dry-run")
+	logrus.Debugf("Creating a wave %s for factory %s targets version %s and new tag %s expires %s",
+		name, factory, version, tag, expires.Format(time.RFC3339))
+
+	new_targets, err := api.TargetsList(factory, version)
```

Comment:
this is WIP, but as you progress the list of targets will need to be something like `GetCurrentProductionTargets + TargetsList(factory, version)`.

Reply:
Yes, sure, see my comment below.